tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_DuDt.cpp

```diff
@@ -0,0 +1,137 @@
+  const auto phi = create_tensor_with_random_values<
+      tnsr::iaa<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto d_psi = create_tensor_with_random_values<
+      tnsr::iaa<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto d_pi = create_tensor_with_random_values<
+      tnsr::iaa<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto d_phi = create_tensor_with_random_values<
+      tnsr::ijaa<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto gauge_function = create_tensor_with_random_values<
+      tnsr::a<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto spacetime_deriv_gauge_function = create_tensor_with_random_values<
+      tnsr::ab<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto gamma0 = create_tensor_with_random_values<Scalar<DataVector>>(
+      n_pts, make_not_null(&gen));
+  const auto gamma1 = create_tensor_with_random_values<Scalar<DataVector>>(
+      n_pts, make_not_null(&gen));
+  const auto gamma2 = create_tensor_with_random_values<Scalar<DataVector>>(
+      n_pts, make_not_null(&gen));
+  const auto lapse = create_tensor_with_random_values<Scalar<DataVector>>(
+      n_pts, make_not_null(&gen));
+  const auto shift = create_tensor_with_random_values<
+      tnsr::I<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto upper_spatial_metric = create_tensor_with_random_values<
+      tnsr::II<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto upper_psi = create_tensor_with_random_values<
+      tnsr::AA<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto christoffel_first_kind = create_tensor_with_random_values<
+      tnsr::abb<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto christoffel_second_kind = create_tensor_with_random_values<
+      tnsr::Abb<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto trace_christoffel_first_kind = create_tensor_with_random_values<
+      tnsr::a<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto normal_one_form = create_tensor_with_random_values<
+      tnsr::a<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+  const auto normal_vector = create_tensor_with_random_values<
+      tnsr::A<DataVector, dim, Frame::Inertial>>(n_pts, make_not_null(&gen));
+
+  GeneralizedHarmonic::ComputeDuDt<dim>::apply(
+      make_not_null(&dt_psi), make_not_null(&dt_pi), make_not_null(&dt_phi),
+      psi, pi, phi, d_psi, d_pi, d_phi, gamma0, gamma1, gamma2, gauge_function,
+      spacetime_deriv_gauge_function, lapse, shift, upper_spatial_metric,
+      upper_psi, trace_christoffel_first_kind, christoffel_first_kind,
+      christoffel_second_kind, normal_vector, normal_one_form);
+
+  CHECK(dt_psi.get(0, 0)[0] == approx(90.741903059323832));
```

Comment:
It would be tedious (but good) to also test DataVectors. Anyone else have thoughts?

Reply:
😕 aren't these DataVectors (of size one)?

Reply:
Ah yes, so we don't want to support doubles is the question then? And we don't want to test DataVectors of realistic sizes?

Reply:
I ask about larger DataVectors since if I recall correctly we've had bugs sneak by us because our DataVectors were too small.

Reply:
Yeah I can test with size 2 DataVectors. I don't think we need the RHS to support doubles since it shouldn't be needed for lensing or anything.